### DISCUSSION TITLE
v0.1.0 roadmap

### DISCUSSION THREAD

**Discussion by aladdin-add** [kataw/kataw#128] [Ideas]
The project is under active development. Just bring it up to give our team and users clarity over our priorities. It's meant more as a set of guidelines than as a strict set of tasks we are bound to finish.

At a high-level, I wish we can get the following goals, and released in v0.1.0:

* Parsing ECMA 262(aka JavaScript), and the cst spec be stable
* Printing API (like prettier)
* `//kataw-ignore`(like `//prettier-ignore`)
* Command line interface (like prettier cli)
* Documentation & website

open questions:
* Do we want to support the real typescript?

**Comment by KFlash** [comment]
Your firsts bullet point is done, but I can double check the cst spec. Regarding the printer it is there, but I need to refactor it to include the transformer code so we can print unique identifiers and get comments correct for transformed CST nodes.

The comment is there and has it purpose?

Regarding CLI. I already put together an internal cli. Only thing left is to get it working for command line.

When it comes to Typescript. I can't see any point why we need to kill the performance? Kataw will include it's own type system and adding things to the parser that is not part of the ECMA spec is out of the scope.

**Reply by aladdin-add** [reply]
re typescript: perf matters, but not everything, I can see very few devs will migrate from typescript, just because of perf. :)

**Reply by KFlash** [reply]
Most devs are using Acorn for parsing. And it can't even parse TS ;)

**Reply by aladdin-add** [reply]
but it has a plugin system, with support jsx/ts/flow. I'm not arguing - it's just a design philosophy, if you put the perf in the first place.

**Comment by KFlash** [comment]
I think you need to add a discussion about a plugin system then? I designed Kataw to be a standalone compiler that should do the same as prettier, babel, and others, but in the moment you enable parsing of type annotations it will turn into it's own language in the same way as Flow and TS. You can't parse TS with Flow either or Flow with TS.

**Reply by aladdin-add** [reply]
sounds good! IMHO, only the JS supports should be built-in, anything else can be a plugin to opt-in.

although it would be a big change.

**Reply by aladdin-add** [reply]
things like https://github.com/kataw/kataw/issues/122 is a very low priority. I suggest focusing the JS supports and plugin system - that's the core.

**Comment by KFlash** [comment]
Right now I'm focusing on writing a new printer, and I gave up indent and adjust of comments. I will assign you to that task soon as I'm done

**Comment by KFlash** [comment]


**Reply by KFlash** [reply]
@aladdin-add You can take a stab at the plugin system and implement it?

**Reply by aladdin-add** [reply]
can be v0.2.0?

**Reply by KFlash** [reply]
yes

**Reply by KFlash** [reply]
@aladdin-add Right now I'm focusing on getting a printer 100% working with comments. After that need to port it to a pretty printer style. `/* 1 */ do /* 2 */{ /* 1 */ } while/* 3 */(x) { }` try that one with Prettier, and the comments are attached wrong :)

**Reply by KFlash** [reply]
I'm finishing the CST specs now while finishing the printer. So two of this bullet points will soon be solved.

**Comment by KFlash** [comment]
CST and the printer is stable now.
